### PR TITLE
LPS-98821 Frontend reads and updates state with editableValues provided by backend

### PR DESCRIPTION
**Pull request by darquesdev**
CC: @andresfulla @ealonso @jkappler

### CODE REVIEW

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddSegmentsExperienceMVCActionCommand.java [hunk 1]
```diff
@@ -186,13 +219,103 @@ private void _populateSegmentsExperienceJSONObject(
 			));
 	}
 
+	private Map<Long, String> _updateFragmentEntryLinksEditableValues(
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddSegmentsExperienceMVCActionCommand.java]: This method traverses editableValues as is https://github.com/liferay/liferay-portal/blob/8dc979e0e6286aa494ff0dc868f4eeac8637d10f/modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/display/context/ContentPageEditorDisplayContext.java#L1255

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddSegmentsExperienceMVCActionCommand.java [hunk 2]
```diff
@@ -156,9 +162,36 @@ private JSONObject _addSegmentsExperience(ActionRequest actionRequest)
...
+
+		for (Map.Entry<Long, String> entry :
+				fragmentEntryLinksEditableValuesMap.entrySet()) {
+
+			fragmentEntryLinksJSONObject.put(
+				String.valueOf(entry.getKey()),
+				JSONFactoryUtil.createJSONObject(entry.getValue()));
+		}
+
+		jsonObject.put("fragmentEntryLinks", fragmentEntryLinksJSONObject);
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddSegmentsExperienceMVCActionCommand.java]: Using the same format as in [EditFragmentEntryLinksMVCActionCommand](https://github.com/liferay/liferay-portal/blob/8dc979e0e6286aa494ff0dc868f4eeac8637d10f/modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/EditFragmentEntryLinksMVCActionCommand.java#L58)

File: modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddSegmentsExperienceMVCActionCommand.java [hunk 3]
```diff
@@ -186,13 +219,103 @@ private void _populateSegmentsExperienceJSONObject(
...
+						editableProcessorKey, editableProcessorJSONObject);
+				}
+			}
+
+			fragmentEntryLinksEditableValuesMap.put(
+				fragmentEntryLink.getFragmentEntryLinkId(),
+				editableValuesJSONObject.toString());
+		}
+
+		_fragmentEntryLinkService.updateFragmentEntryLinks(
```
Comment (darquesdev) [inline modules/apps/layout/layout-content-page-editor-web/src/main/java/com/liferay/layout/content/page/editor/web/internal/portlet/action/AddSegmentsExperienceMVCActionCommand.java]: Using the recently created method [updateFragmentEntryLinks](https://github.com/liferay/liferay-portal/blob/8dc979e0e6286aa494ff0dc868f4eeac8637d10f/modules/apps/fragment/fragment-service/src/main/java/com/liferay/fragment/service/impl/FragmentEntryLinkServiceImpl.java#L132)

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
ci:test:sf

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by epgarcia** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by epgarcia** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/76580. <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**